hack/update/golang_version/update_golang_version.go

```diff
@@ -36,11 +36,6 @@ var (
 	}
 
 	schema = map[string]update.Item{
-		"go.mod": {
```

Comment:
for the go.mod  update, what does Kubernetes project do ? it would be nice to be in Par with them.

Reply:
It seems they just let it auto update when the underlying packages require it, if an underlying package requires it and you run `go mod tidy` it will change the version automatically, we could follow that.

Reply:
that sounds reasonable